### DISCUSSION TITLE
Is it possible to have an animated gif as a source to draw lines?

### DISCUSSION THREAD

**Discussion by barathi-k** [pubpub-zz/ppInk#263] [Ideas]
Firstly, thanks for the wonderful tool. 

I was wondering if there is a way we would take an animated gif file like this :
![1](https://github.com/pubpub-zz/ppInk/assets/55287040/d2fe86c2-35ca-4670-b5b1-f0e04533935e)
and use it to draw a freehand line like this:

https://github.com/pubpub-zz/ppInk/assets/55287040/903b5ec3-03bd-4d1c-90ca-24cbd2f05887

Thanks!

**Comment by pubpub-zz** [comment]
Hi @barathi-k 
    Happy you enjoy.
    We should get the good rendering using the line of pattern with some slight adjustments. To complete my test, I would need an animated line same width but with a length a forth of the one you've attached to your message. Can you provide it ?

**Comment by barathi-k** [comment]
Thanks for your quick response. Here is a smaller version. Would be very happy to provide any other variations or totally new images if you need.

Please find below.
![small](https://github.com/pubpub-zz/ppInk/assets/55287040/5abb562d-b1d2-4fed-a747-3bd3df675037)

**Comment by pubpub-zz** [comment]
here you are my rendering proposal:
[ppInk slider.zip](https://github.com/pubpub-zz/ppInk/files/11712277/ppInk.slider.zip)

To use it:
* go to the clipart button,
* select the icon. You can note that the filename starts with ~ : this will be the way to identify images that will be rotated to follow the line
* select the "line of pattern" in the list at the bottom of the window
* the cursor become a target (arrow in a circle) : as before, select the size of the image : It will be a solution to slight adjust the size (mainly width) of the image
* Select the distance between two images: you should set it in order to set the two images to sligthly overlap
* then draw the path : the images are not rotated during the drawing but after only

The rendering looks good for me

**Comment by barathi-k** [comment]
Works amazing!  

One last request. Can we have an api call that sets the image size and the overlap so that we are ready to start drawing at the press of a hotkey? 

Not a request (may be a suggestion): I might be going crazy here: Would it be better if we start with zero opacity for all these images and then make them fully visible after they are rotated to their right orientation? And to give a sense of animation/drawing, make them appear in the order they were created with a tiny bit of delay between each appearing. I understand people might think that the drawing is lagging with this approach. Just thinking loud.

**Reply by pubpub-zz** [reply]
> Works amazing!

😀😀😀

> One last request. Can we have an api call that sets the image size and the overlap so that we are ready to start drawing at the press of a hotkey?

you have already the feature if you check save pattern setup: the first pattern will be saved the parameters. this works also with the 3 predefined image stamps.

> Not a request (may be a suggestion): I might be going crazy here: Would it be better if we start with zero opacity for all these images and then make them fully visible after they are rotated to their right orientation? And to give a sense of animation/drawing, make them appear in the order they were created with a tiny bit of delay between each appearing. I understand people might think that the drawing is lagging with this approach. Just thinking loud.

The image are now drawn rotated. this should be better. That's the best I can propose
the new version:
[ppInk slider 2.zip](https://github.com/pubpub-zz/ppInk/files/11714187/ppInk.slider.2.zip)

**Reply by barathi-k** [reply]
Amaze Balls! I just got my API call successfully!! Thanks a lot. The image rotation isn't working on my PC though. But I just love this new toy for now and I'm going to experiment with more animated arrows and forget about the roatation. Thanks a lot!

**Reply by barathi-k** [reply]
Just to give you some context: 
The problem with rotation is only when called through the API before I manually set it once. It works otherwise. 
Here is my API call : http://localhost:7999/CurrentTool?T=23&F=-1&I=~glidding_arrow&W=30&H=20&D=600

All works good now.

**Reply by pubpub-zz** [reply]
Got it!
The flag was forgotten in the REST API. I did not understood at the beginning you were using it!
This should be good
[ppInk Glidder3.zip](https://github.com/pubpub-zz/ppInk/files/11714630/ppInk.Glidder3.zip)

**Reply by barathi-k** [reply]
Works falwless! thanks a lot.
